### PR TITLE
fix: adding offline error for git failure

### LINKED ISSUE #27589: [Bug]: Airgapped instance: Git modal doesn't throw any error when connection is not successful
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

On trying to connect app to git when connection is not successful, there is no error message shown for it:
Screenshot 2023-09-25 at 12 46 35 PM

### Steps To Reproduce

1. Create an app and connect to it to git by adding repo URL
2. Do not add deploy key in repository and click on connect button
3. Observe there is no error thrown for it, ideally it should throw error stating: SSH key is not configured correctly

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Airgap: release

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Error message were ignored when offline. Fixed it!

#### PR fixes following issue(s)
Fixes #27589

#### Media
image

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by Parthvi12** [comment]
Since this cannot be tested on DP will be verifying fix on Airgapped image. Also @brayn003 has verified the fix in his local, good to go from my side.

File: app/client/src/sagas/GitSyncSagas.ts [hunk 1]
```diff
@@ -248,11 +248,14 @@ function* connectToGitSaga(action: ConnectToGitReduxAction) {
 
     // Api error
     // Display on the UI
-    if (response && !response?.responseMeta?.success) {
+
+    const errorResponse = response || error?.response.data;
```
Comment (sneha122) [inline app/client/src/sagas/GitSyncSagas.ts:252]: same here as well, can we make it response?.data?

**Review by sneha122 — APPROVED** [review]

**Pull request merged**